tests/integration.rs

```diff
@@ -0,0 +1,495 @@
+        info!("testing HTTP-01 challenge");
+
+        debug!("creating order for example.com");
+        // Create an order.
+        let identifier = Identifier::Dns("example.com".to_owned());
+        let mut order = account
+            .new_order(&NewOrder {
+                identifiers: &[identifier],
+            })
+            .await?;
+        info!("created order {}", order.url());
+
+        let authorizations = order.authorizations().await?;
+        let mut challenges = Vec::with_capacity(authorizations.len());
+        let mut names = Vec::with_capacity(authorizations.len());
+
+        // Collect up the relevant challenges, provisioning the expected responses as we go.
+        for authz in &authorizations {
+            match authz.status {
+                AuthorizationStatus::Pending => {}
+                AuthorizationStatus::Valid => continue,
+                _ => unreachable!("unexpected authz state: {:?}", authz.status),
+            }
+
+            let challenge = authz
+                .challenges
+                .iter()
+                .find(|c| c.r#type == ChallengeType::Http01)
+                .ok_or_else(|| "no http01 challenge found")?;
+
+            let Identifier::Dns(identifier) = &authz.identifier;
+
+            // Provision the HTTP-01 challenge response with the Pebble challenge test server.
+            let key_auth = order.key_authorization(challenge);
+            debug!(
+                "provisioning HTTP-01 response: token={token} key_auth={key_auth}",
+                token = challenge.token,
+                key_auth = key_auth.as_str()
+            );
+            self.add_http01_response(&challenge.token, key_auth.as_str())
+                .await?;
+
+            challenges.push((identifier, &challenge.url));
+            names.push(identifier.clone());
+        }
+
+        // Tell the CA we have provisioned the response for each challenge.
+        for (_, url) in &challenges {
+            debug!("marking challenge ready: {url}");
+            order.set_challenge_ready(url).await?;
+        }
+
+        // Poll until the order is ready.
+        Self::poll_until_ready(&mut order).await?;
+
+        // Issue a certificate for the names and test the chain validates to the issuer root.
+        self.issue_certificate(&mut order, names).await
+    }
+
+    /// Provision an HTTP-01 challenge response for the given token and key authorization.
+    ///
+    /// The Pebble challenge test server will be configured to respond to HTTP-01 challenge
+    /// requests for the provided token by returning the expected key auth.
+    async fn add_http01_response<'token, 'key_auth>(
+        &self,
+        token: &'token str,
+        key_auth: &'key_auth str,
+    ) -> Result<(), Box<dyn StdError>> {
+        #[derive(Serialize)]
+        struct AddHttp01Request<'token, 'key_auth> {
+            token: &'token str,
+            #[serde(rename = "content")]
+            key_auth: &'key_auth str,
```

Comment:
Nit: let's just call the field `content`, please.

(But maybe consider inlining the `body` binding, which feels fairly trivial?)

Reply:
Done x2.